connector/nopconnector/nop.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nopconnector // import "go.opentelemetry.io/collector/connector/nopconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/internal/sharedcomponent"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	typeStr = "nop"
```

Comment:
I think the name is confusing. I thought "nop" is something like /dev/null, a blackhome, while in reality it is a perfect passthrough.

Reply:
As I understand it, the meaning of [nop](https://en.wikipedia.org/wiki/NOP_(code)#Machine_language_instructions) is basically "passthrough". Regardless, I have reservations about the name for another reason. 

Basically, I think users will be less interested in the passthrough behavior than the fact that the connector can be used to fork and merge pipelines. Technically, the fork/merge behavior comes from the connector framework itself, but ultimately this connector will be the one used when forking/merging should be done without altering the data.

I think maybe `hub` is a better name, as it conveys something more directly related to the use cases it support. WDYT?

Reply:
I would also prefer something more descriptive than `nop`.  `hub` could work.  I also like `link` or `join`.

Reply:
I don't care for `join` because it does convey the ability to fork pipelines. 

`link` is a bit better, but in my mind this conveys a 1:1 connection between pipelines. 

Another option might be `multiplex`, but this feels less accessible than `hub`.

Reply:
One more option - `forward`. I think this is intuitive for both fork and merge cases, and does not imply any alteration of the data.

```yaml
service:
  pipelines:
    logs/in:
      receivers: [filelog]
      exporters: [forward]
    logs/out1:
      receivers: [forward]
      exporters: [backend/1]
    logs/out2:
      receivers: [forward]
      exporters: [backend/2]
```

```yaml
service:
  pipelines:
    logs/in1:
      receivers: [filelog]
      exporters: [forward]
    logs/in2:
      receivers: [filelog]
      exporters: [forward]
    logs/out:
      receivers: [forward]
      exporters: [backend]
```